### PR TITLE
build: Bump minimum required C++ standard to 20

### PR DESCRIPTION
**Pull request by paulgessinger**
Blocked by:

- #3227
- https://github.com/acts-project/machines/pull/113

### CODE REVIEW

**Comment by stephenswat** [comment]
_Yes_ :fire:

**Comment by paulgessinger** [comment]
Fyi @CarloVarni

**Comment by paulgessinger** [comment]
physmon changes seem consistent with statistical variation.

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
             ${{ runner.os  }}-ccache-linux_ubuntu_${{ env.CCACHE_KEY_SUFFIX }}_
 
       - name: Configure
-        # setting CMAKE_CXX_STANDARD=17 is a workaround for a bug in the
+        # setting CMAKE_CXX_STANDARD=20 is a workaround for a bug in the
```
Comment (andiwand) [inline .github/workflows/builds.yml:50]: still valid?

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -269,7 +267,7 @@ jobs:
 
 
       - name: Configure
-        # setting CMAKE_CXX_STANDARD=17 is a workaround for a bug in the
+        # setting CMAKE_CXX_STANDARD=20 is a workaround for a bug in the
```
Comment (andiwand) [inline .github/workflows/builds.yml:270]: same here

File: .github/workflows/builds.yml [hunk 3]
```diff
@@ -367,7 +365,7 @@ jobs:
             ${{ runner.os  }}-ccache_${{ env.CCACHE_KEY_SUFFIX }}_
 
       - name: Configure
-        # setting CMAKE_CXX_STANDARD=17 is a workaround for a bug in the
+        # setting CMAKE_CXX_STANDARD=20 is a workaround for a bug in the
```
Comment (andiwand) [inline .github/workflows/builds.yml:368]: same here

File: Examples/Python/tests/root_file_hashes.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-test_pythia8__pythia8_particles.root: fb7d886314f737847838a714a4aa49c179cfaabfd9f16e3d4b7672a7d12db0d3
+test_pythia8__pythia8_particles.root: 99085d06a4626d39756283d1b4c3e502d667b2b6fb04be54d09dd599404caca7
```
Comment (andiwand) [inline Examples/Python/tests/root_file_hashes.txt:1]: Is this just a pythia8 version change?

**Review by andiwand — APPROVED** [review]
noiiiiice

File: Examples/Python/tests/root_file_hashes.txt (hunk 4 above)
Comment (paulgessinger) [reply Examples/Python/tests/root_file_hashes.txt:1]: I believe so, yes.

File: .github/workflows/builds.yml (hunk 1 above)
Comment (paulgessinger) [reply .github/workflows/builds.yml:50]: Not sure, but I'll investigate this separately. For now I'm keeping it explicit in the CI config.

**Review by andiwand — APPROVED** [review]

**Comment by paulgessinger** [comment]
Let's try again!

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [41c5a73a2b669ea5889d79c8a7a88318d6e06244]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/41161564) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7744488)!

**Pull request merged**